src/SWP/Bundle/CoreBundle/Provider/ContainerProvider.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Superdesk Web Publisher Core Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\CoreBundle\Provider;
+
+use Doctrine\ORM\Query;
+use SWP\Bundle\RevisionBundle\RevisionContext;
+use SWP\Bundle\TemplatesSystemBundle\Provider\ContainerProvider as BaseContentProvider;
+use SWP\Component\Common\Criteria\Criteria;
+use SWP\Component\TemplatesSystem\Gimme\Model\ContainerInterface;
+
+/**
+ * Revisions Aware ContainerProvider.
+ */
+class ContainerProvider extends BaseContentProvider
+{
+    /**
+     * @var RevisionContext
+     */
+    protected $revisionContext;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getQueryForAll(): Query
+    {
+        $criteria = new Criteria();
+        $criteria->set('revision', $this->revisionContext->getCurrentRevision());
+
+        return $this->containerRepository->getQueryByCriteria($criteria, [], 'r')->getQuery();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOneByName(string $name)
+    {
+        $qb = $this->containerRepository->getByName($name);
+        $qb->andWhere('c.revision = :revision')
+            ->setParameter('revision', $this->revisionContext->getCurrentRevision());
+
+        return $qb->getQuery()->getOneOrNullResult();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOneById($uuid)
+    {
+        $qb = $this->containerRepository->createQueryBuilder('c')
```

Comment:
this query should go to the container repository into a method: e.g.  `findContainerByIdAndCurrentRevision`